Core/Node-API/Source/js_native_api_v8.cc

```diff
@@ -2563,68 +2931,142 @@ napi_status napi_create_arraybuffer(napi_env env,
+namespace v8impl {
 
-  if (finalize_cb != nullptr) {
-    // Create a self-deleting weak reference that invokes the finalizer
-    // callback.
-    v8impl::Reference::New(env,
-        buffer,
-        0,
-        true,
-        finalize_cb,
-        external_data,
-        finalize_hint);
-  }
+    namespace {
+
+        class BufferFinalizer : private Finalizer {
+        public:
+            static BufferFinalizer* New(napi_env env,
+                napi_finalize finalize_callback = nullptr,
+                void* finalize_data = nullptr,
+                void* finalize_hint = nullptr) {
+                return new BufferFinalizer(
+                    env, finalize_callback, finalize_data, finalize_hint);
+            }
+            // node::Buffer::FreeCallback
+            static void FinalizeBufferCallback(char* data, void* hint) {
+                std::unique_ptr<BufferFinalizer, Deleter> finalizer{
+                    static_cast<BufferFinalizer*>(hint) };
+                finalizer->finalize_data_ = data;
+
+                // It is safe to call into JavaScript at this point.
+                if (finalizer->finalize_callback_ == nullptr) return;
+                finalizer->env_->CallFinalizer(finalizer->finalize_callback_,
+                    finalizer->finalize_data_,
+                    finalizer->finalize_hint_);
+            }
+
+            struct Deleter {
+                void operator()(BufferFinalizer* finalizer) { delete finalizer; }
+            };
+
+        private:
+            BufferFinalizer(napi_env env,
+                napi_finalize finalize_callback,
+                void* finalize_data,
+                void* finalize_hint)
+                : Finalizer(env, finalize_callback, finalize_data, finalize_hint) {
+                env_->Ref();
+            }
+
+            ~BufferFinalizer() { env_->Unref(); }
+        };
 
-  *result = v8impl::JsValueFromV8LocalValue(buffer);
-  return GET_RETURN_STATUS(env);
-}
+    }
+} // v8impl
+
+
+napi_status NAPI_CDECL
+napi_create_external_buffer(napi_env env,
+    size_t length,
+    void* data,
+    napi_finalize nogc_finalize_cb,
+    void* finalize_hint,
+    napi_value* result) {
+    napi_finalize finalize_cb = reinterpret_cast<napi_finalize>(nogc_finalize_cb);
+    NAPI_PREAMBLE(env);
+    CHECK_ARG(env, result);
 
-napi_status napi_get_arraybuffer_info(napi_env env,
-                                      napi_value arraybuffer,
-                                      void** data,
-                                      size_t* byte_length) {
-  CHECK_ENV(env);
+#if defined(V8_ENABLE_SANDBOX)
+    return napi_set_last_error(env, napi_no_external_buffers_allowed);
+#endif
+
+    v8::Isolate* isolate = env->isolate;
+
+    // The finalizer object will delete itself after invoking the callback.
+    v8impl::BufferFinalizer* finalizer =
+        v8impl::BufferFinalizer::New(env, finalize_cb, nullptr, finalize_hint);
+
+    v8::MaybeLocal<v8::Object> maybe = {};
```

Comment:
internal buffer is not used in JSRuntimeHost, only used in BabylonNative. So, not an issue to get CI tests to pass. Finding the minimal set of code to get `node::Buffer::New` to work is difficult.

Reply:
Is it something that can be fixed with NODE_SPECIFIC ?